pkgs/os-specific/linux/numactl/default.nix

```diff
@@ -17,6 +17,8 @@ stdenv.mkDerivation rec {
     patchShebangs test
   '';
 
+  LDFLAGS = lib.optionalString stdenv.hostPlatform.isRiscV "-latomic";
```

Comment:
You are sure this should not be targetPlatform?

Reply:
The targetPlatform only really makes sense for compilers which might "target" (build executables for) a different platform from the one they're running on. AFAIK this isn't the case for numactl, which is just library and a set of utilities for the host platform.